bitcoin/src/blockdata/transaction.rs

```diff
@@ -512,15 +513,15 @@ impl TxOut {
         let dust_amount = (len as u64) * 3;
 
         TxOut {
-            value: dust_amount + 1, // minimal non-dust amount is one higher than dust amount
+            value: Amount::from_sat(dust_amount + 1), // minimal non-dust amount is one higher than dust amount
             script_pubkey,
         }
     }
 }
 
 // This is used as a "null txout" in consensus signing code.
 impl Default for TxOut {
-    fn default() -> TxOut { TxOut { value: 0xffffffffffffffff, script_pubkey: ScriptBuf::new() } }
+    fn default() -> TxOut { TxOut { value: Amount::from_sat(0xffffffffffffffff), script_pubkey: ScriptBuf::new() } }
```

Comment:
Unrelated but looks like this should be an associated constant `TxOut::NULL`

Reply:
Agreed. And we probably shouldn't use `default` to construct it because it's actually a pretty obscure thing.

Reply:
I don't think this can be made an associated constant because `ScriptBuf::new()` is not a `const`.

```
impl TxOut {
    pub const NULL: Self = TxOut { 
        value: Amount::from_sat(0xffffffffffffffff), 
        script_pubkey: ScriptBuf::new() } }
    }
    ..
}
```

Is there a reason this needs to be an associated `const`?

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/pull/1838